xonsh/built_ins.py

```diff
@@ -730,7 +731,9 @@ def resolve_executable_commands(self):
         if self.binary_loc is None:
             return
         try:
-            self.cmd = get_script_subproc_command(self.binary_loc, self.cmd[1:])
+            scriptcmd = get_script_subproc_command(self.binary_loc, self.cmd[1:])
```

Comment:
Maybe the right thing to do is to pass in `alias` here and then in `get_script_subproc_command()`, rather than returning None, return `[alias] + args`

Reply:
Nope, I don't think so. The point is, that there's a difference between running `ls` and `/bin/ls`. It's not related to `alias`.